core/src/main/java/org/mule/VoidMuleEvent.java

```diff
@@ -40,7 +40,12 @@ public static VoidMuleEvent getInstance()
         return instance;
     }
 
-    private VoidMuleEvent()
+    public static boolean isVoid(MuleEvent event)
```

Comment:
The reason I added this wasn't to avoid == check, but rather because there are cases I need to check if the event is VoidMuleEvent *or* NonBlockingVoidMuleEvent and didn't really want to do if(event == VoidMuleEvent.getInstance() || event == NonBlockingVoidMuleEvent.getInstance()) or event instance VoidMuleEvent.  I can switch back to instanceof test if you think it is preferred though.